### PR TITLE
MM-34444 Detox/E2E: Prepare plugin API and webhook server

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- Added Plugins API
    - `detox/e2e/support/server_api/plugin.js`
- Added webhook server
    - `detox/e2e/plugins/post_message_as.js`
    - `detox/e2e/utils/webhook_utils.js`
    - `detox/webhook_server.js`

1. Feature flags via split.io - I don't think this is implemented in Cypress e2e just yet; @saturninoabril may have a better idea to implement in both Cypress and Detox e2e
2. Apps proxy plugin installed and enabled --> this is achieved in Cypress via MM API calls; the same is done for Detox
3. webhook server --> similar to Cypress webhook server.

In summary, the feature flag capability is not yet available and any new detox tests may need to be disabled until the feature flag is permanently enabled.

#### Ticket Link
JIRA: https://mattermost.atlassian.net/browse/MM-34444

### CODE REVIEW

**Review by larkox — APPROVED** [review]
LGTM. Thank you very much Joseph!

File: detox/package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "axios": "0.21.1",
     "babel-jest": "26.6.3",
     "babel-plugin-module-resolver": "4.1.0",
+    "client-oauth2": "4.3.3",
```
Comment (saturninoabril) [inline detox/package.json:14]: Are we going to use the upstream package or the fork from Daniel? In Cypress is like [this](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/package.json#L12).

File: detox/webhook_server.js [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+    const response = {
+        text: getWebhookResponse(body, {responseType, username, iconUrl}),
+        username,
+        icon_url: iconUrl,
+        type: '',
+        response_type: responseType,
+    };
+    res.status(200).send(response);
+}
```
Comment (saturninoabril) [inline detox/webhook_server.js:296]: Looks good to me but maybe in the future, let's extract this into its own repo/package so it can be shared not only between webapp and mobile, but other repos too, like plugins.

**Review by saturninoabril — APPROVED** [review]
Thanks Joseph! LGTM, except for couple query and pointer.

File: detox/package.json (hunk 1 above)
Comment (josephbaylon) [reply detox/package.json:14]: @saturninoabril I'm not sure. That would be a question for @larkox .

@larkox Please let me know if you need me to use the fork you created.

File: detox/webhook_server.js (hunk 2 above)
Comment (josephbaylon) [reply detox/webhook_server.js:296]: @saturninoabril Yeah, we can definitely extract this in the future.

File: detox/package.json (hunk 1 above)
Comment (josephbaylon) [reply detox/package.json]: Done
